src/providers/be_dyndns.c

```diff
@@ -355,46 +358,78 @@ static uint8_t *nsupdate_convert_address(struct sockaddr_storage *add_address)
     return addr;
 }
 
-static char *nsupdate_msg_add_ptr(char *update_msg,
-                                  struct sockaddr_storage *address,
-                                  const char *hostname,
-                                  int ttl,
-                                  bool delete)
+static char *
+nsupdate_msg_add_ptr(char *update_msg, struct sss_iface_addr *addresses,
+                     const char *hostname, int ttl, uint8_t remove_af,
+                     bool update_per_family)
 {
-    char *strptr;
+    char *updateipv4 = talloc_strdup(update_msg, "");
+    char *updateipv6 = talloc_strdup(update_msg, "");
+    char *ptr;
+    struct sss_iface_addr *address_it;
     uint8_t *addr;
 
-    addr = nsupdate_convert_address(address);
-    if (addr == NULL) {
+    if (!updateipv4 || !updateipv6) {
         return NULL;
     }
-
-    strptr = resolv_get_string_ptr_address(update_msg, address->ss_family,
-                                           addr);
-    if (strptr == NULL) {
-        return NULL;
+    DLIST_FOR_EACH(address_it, addresses) {
+        addr = nsupdate_convert_address(address_it->addr);
+        if (addr == NULL) {
+            return NULL;
+        }
+        ptr = resolv_get_string_ptr_address(update_msg, address_it->addr->ss_family,
+                                            addr);
+        if (ptr == NULL) {
+            return NULL;
+        }
+        switch (address_it->addr->ss_family) {
+        case AF_INET:
+            if (remove_af & DYNDNS_REMOVE_A) {
+                updateipv4 = talloc_asprintf_append(updateipv4,
+                                                    "update delete %s in PTR\n",
+                                                    ptr);
+            }
+            if (updateipv4) {
+                updateipv4 = talloc_asprintf_append(updateipv4,
+                                                    "update add %s %d in PTR %s.\n",
+                                                    ptr, ttl, hostname);
+            }
+            break;
```

Comment:
It would read better if it is:
```c
if (remove...) {
    updateipv4 = talloc...
    if (updateipv4 == NULL) {
        return NULL;
    }
}

updateipv4 = talloc...
if (updateipv4 == NULL) {
    return NULL;
}

Same for ipv6 case.
```

Reply:
done